jrt/src/com/yahoo/jrt/slobrok/api/Register.java

```diff
@@ -191,16 +192,21 @@ private void handleUpdate() {
                     lastRegisterSucceeded.remove(name);
                 } else {
                     final Boolean lastSucceeded = lastRegisterSucceeded.get(name);
-                    if (lastSucceeded == null || lastSucceeded != !req.isError()) {
+                    if (lastSucceeded == null) {
+                        logOnSuccess = true;
+                        logOnFailure = false;
+                    } else if (lastSucceeded != !req.isError()) {
                         logOnSuccess = true;
-                        lastRegisterSucceeded.put(name, !req.isError());
                     }
+                    lastRegisterSucceeded.put(name, !req.isError());
```

Comment:
This method is now called unconditionally wrt. `lastSucceeded`, can this have any unintended side effects?

Reply:
> This method is now called unconditionally wrt. lastSucceeded, can this have any unintended side effects?

no - if it already had the value from before, we will overwrite with same value, to avoid duplicating this line.